source/renderer/app/components/staking/delegation-setup-wizard/DelegationSetupWizardDialog.js

```diff
@@ -0,0 +1,72 @@
+// @flow
+import React, { Component } from 'react';
+import { observer } from 'mobx-react';
+import DelegationStepsNotAvailableDialog from './DelegationStepsNotAvailableDialog';
+import DelegationStepsIntroDialog from './DelegationStepsIntroDialog';
+import DelegationStepsChooseWalletDialog from './DelegationStepsChooseWalletDialog';
+
+type WalletData = {
+  label: string,
+  value: string,
+  isAcceptableSetupWallet: boolean,
+};
+
+type Props = {
+  isDisabled: boolean,
+  activeStep: number,
+  wallets: Array<WalletData>,
+  onClose: Function,
+  onContinue: Function,
+  onBack: Function,
+  onLearnMoreClick: Function,
+  stepsList: Array<string>,
+  minDelegationFunds: number,
+};
+
+@observer
+export default class DelegationSetupWizardDialog extends Component<Props> {
+  render() {
+    const {
+      isDisabled,
+      onClose,
+      onContinue,
+      onBack,
+      activeStep,
+      wallets,
+      onLearnMoreClick,
+      stepsList,
+      minDelegationFunds,
+    } = this.props;
+
+    let content = null;
+    if (isDisabled) {
+      content = (
+        <DelegationStepsNotAvailableDialog
+          minDelegationFunds={minDelegationFunds}
+          onClose={onClose}
+        />
+      );
+    } else if (activeStep === 0) {
```

Comment:
@tomothespian please change these else-if's into case structure - that will be much more clean and useful for the rest of the steps 🙏

Reply:
@tomothespian can you please change this too "case" structure? 🙏